### PR TITLE
[DataObject] Localized-Field Relation from Field-Collection in Data Object cannot be saved 

### LINKED ISSUE #5719: [DataObject] Localized-Field Relation from Field-Collection in Data Object cannot be saved
# Bug Report

### Expected behavior
Update a existing releation, instead catch the exception from insert in logger.

### Actual behavior
Throw exception and shows validation failed "unique constraint violation" in backend.
SQLSTATE[23000]: Integrity constraint violation: 1062 Duplicate entry '4-3-localizedfield-/fieldcollection~footerColumn1/0/localizedfie' for key 'PRIMARY'

### Steps to reproduce
Add a field collection with a localized field with a many to one field and use it as footerColumn1... in a class like:
image
Create a new object and drag a document to the many to one field, save the object -> press the reload button and click on save & publish.

### Code example
use
```php
$db->insertOrUpdate('object_relations_' . $classId, $relation);
```
instead of insert exception:
```php
$db->insert('object_relations_' . $classId, $relation);
```

### PR DESCRIPTION
**Pull request by albertmueller**
## Changes in this pull request  
Resolves #5719

### CODE REVIEW

**Comment by brusch** [comment]
@albertmueller thanks for your PR, unfortunately it seems that this change causes some issues, see: https://travis-ci.org/pimcore/pimcore/jobs/646321883#L4359 
Thanks for checking  🙏

**Comment by weisswurstkanone** [comment]
Hi @albertmueller , do non-localized relations still work with your change? Could you please check that?

Besides that, would it be possible for you to add a unit test that demonstrates the problem?

**Comment by albertmueller** [comment]
@weisswurstkanone every relation type works well localized and non localized. the issue is only reproducible via the ui backend, have a look at #5719

**Comment by weisswurstkanone** [comment]
Yes, saw that, but it talks about localized relations inside fieldcollections.
So my question is if normal relations inside field collections are still ok.

And: unit test would be cool.

**Comment by albertmueller** [comment]
@weisswurstkanone do you provide somewhere a docker image for codeception / testing? because i can't run the tests in my php 7.3 docker container, got an error:
In Pimcore.php line 247:
  [Symfony\Component\Debug\Exception\FatalThrowableError]  
  Call to a member function get() on null
Is there a secial setup required?

**Comment by weisswurstkanone** [comment]
@dpfaffenbauer ,  can you give some advice?

**Comment by dpfaffenbauer** [comment]
the tests can run in the container, I already did that several times. Not sure why it's not working for albert

**Comment by weisswurstkanone** [comment]
Did you run them with error_reporting E_ALL  as outlined here ?

https://pimcore.com/docs/6.x/Development_Documentation/Development_Tools_and_Details/Testing/Core_Tests.html

**Comment by albertmueller** [comment]
@weisswurstkanone the tests are now running (it was my mistake, had some wrong repos including codecp), but im not able to reproduce the error from travis:
1) AdvancedManyToManyRelationTest: Field collection localized attributes
Test  tests/model/LazyLoading/AdvancedManyToManyRelationTest.php:testFieldCollectionLocalizedAttributes
[PHPUnit\Framework\Exception] Undefined index: metadata

my test result (PHP 7.3.13 / Symfony 4.4.4):
✔ AdvancedManyToManyObjectRelationTest: Class attributes (1.88s)
✔ AdvancedManyToManyObjectRelationTest: Localized class attributes (0.56s)
✔ AdvancedManyToManyObjectRelationTest: Block class attributes (0.40s)
✔ AdvancedManyToManyObjectRelationTest: Lazy block class attributes (0.45s)
✔ AdvancedManyToManyObjectRelationTest: Field collection attributes (0.45s)
✔ AdvancedManyToManyObjectRelationTest: Field collection localized attributes (0.45s)
✔ AdvancedManyToManyObjectRelationTest: Brick attributes (0.69s)
✔ AdvancedManyToManyObjectRelationTest: Localized brick attributes (0.95s)
✔ AdvancedManyToManyRelationTest: Class attributes (0.46s)
✔ AdvancedManyToManyRelationTest: Localized class attributes (0.50s)
✔ AdvancedManyToManyRelationTest: Block class attributes (0.48s)
✔ AdvancedManyToManyRelationTest: Lazy block class attributes (0.53s)
✔ AdvancedManyToManyRelationTest: Field collection attributes (0.42s)
✔ AdvancedManyToManyRelationTest: Field collection localized attributes (0.44s)
✔ AdvancedManyToManyRelationTest: Brick attributes (0.46s)
✔ AdvancedManyToManyRelationTest: Localized brick attributes (0.65s)
✔ ManyToManyObjectRelationTest: Class attributes (0.42s)
✔ ManyToManyObjectRelationTest: Localized class attributes (0.41s)
✔ ManyToManyObjectRelationTest: Block class attributes (0.55s)
✔ ManyToManyObjectRelationTest: Lazy block class attributes (0.42s)
✔ ManyToManyObjectRelationTest: Field collection attributes (0.41s)
✔ ManyToManyObjectRelationTest: Field collection localized attributes (0.45s)
✔ ManyToManyObjectRelationTest: Brick attributes (0.67s)
✔ ManyToManyObjectRelationTest: Localized brick attributes (0.85s)
✔ ManyToManyRelationTest: Class attributes (0.42s)
✔ ManyToManyRelationTest: Localized class attributes (0.46s)
✔ ManyToManyRelationTest: Block class attributes (0.46s)
✔ ManyToManyRelationTest: Lazy block class attributes (0.46s)
✔ ManyToManyRelationTest: Field collection attributes (0.58s)
✔ ManyToManyRelationTest: Field collection localized attributes (0.44s)
✔ ManyToManyRelationTest: Brick attributes (0.47s)
✔ ManyToManyRelationTest: Localized brick attributes (0.66s)
✔ ManyToOneRelationTest: Class attributes (0.38s)
✔ ManyToOneRelationTest: Localized class attributes (0.45s)
✔ ManyToOneRelationTest: Block class attributes (0.54s)
✔ ManyToOneRelationTest: Lazy block class attributes (0.40s)
✔ ManyToOneRelationTest: Field collection attributes (0.43s)
✔ ManyToOneRelationTest: Field collection localized attributes (0.48s)
✔ ManyToOneRelationTest: Brick attributes (0.53s)
✔ ManyToOneRelationTest: Localized brick attributes (0.70s)
✔ MultipleAssigmentTest: Multiple assignments on single many to many (0.73s)
✔ MultipleAssigmentTest: Multiple assignments on single many to many object (0.31s)
✔ MultipleAssigmentTest: Multiple assignments multiple many to many (0.45s)
✔ MultipleAssigmentTest: Multiple assignments multiple many to many object (0.31s)

can you please check the PR on your machine?

**Comment by weisswurstkanone** [comment]
did you see my comment about error reporting ?

**Comment by albertmueller** [comment]
yes and what should that be?

**Comment by weisswurstkanone** [comment]
make sure that error reporting is set to E_ALL, otherwise you will experience a different behavior when running the locally

**Comment by albertmueller** [comment]
it is on and i tryed it also with -vvv, but i got on this tests no error, that's why i don't understand that travis shows there an error. i thinkt "Undefined index: metadata" has nothing to do with my change from the PR.

**Comment by weisswurstkanone** [comment]
Still think that something is wrong with your error level as I can see the error when I run it locally

File: models/DataObject/Traits/ContextPersistenceTrait.php [hunk 1]
```diff
@@ -57,8 +57,8 @@ protected function enrichDataRow($object, array $params, ?string &$classId, &$ro
             $context = $object->getContext();
             if (isset($context['containerType']) && ($context['containerType'] === 'fieldcollection' || $context['containerType'] === 'objectbrick')) {
                 $fieldname = $context['fieldname'];
-                $index = $context['index'] ?? null;
-                $row['ownername'] = '/' . $context['containerType'] . '~' . $fieldname . '/' . $index . '/localizedfield~' . $row['ownername'];
+                $row['index'] = $context['index'] ?? 0;
```
Comment (weisswurstkanone) [inline models/DataObject/Traits/ContextPersistenceTrait.php]: This is definitely wrong for objectbricks as there we don't have an index.
Previous: "//"
After: "/0/"

Comment (albertmueller) [reply models/DataObject/Traits/ContextPersistenceTrait.php]: thx for review, then it's better to keep the index as it was before. if i have more time then i will setup a better docker image for testing.

**Comment by weisswurstkanone** [comment]
already fixed via https://github.com/pimcore/pimcore/pull/5749

**Pull request closed without merging**